### PR TITLE
[sw] separate clock frequency enums

### PR DESCRIPTION
**Pull request by tjaychen**
- This is needed to for divided clocks in the design. See #3211

Signed-off-by: Timothy Chen <timothytim@google.com>

### CODE REVIEW

**Comment by tjaychen** [comment]
hey all, please have a look.

I kind of wonder if it makes sense long term to have another wrapping function that initializes uart...since so many tests call it.  In the future we if change the enum again, we'd have to update a pile of files.

File: sw/device/lib/arch/device_fpga_nexysvideo.c [hunk 1]
```diff
@@ -11,7 +11,11 @@
 
 const device_type_t kDeviceType = kDeviceFpgaNexysVideo;
 
-const uint64_t kClockFreqHz = 50 * 1000 * 1000;  // 50MHz
+const uint64_t kClockFreqCpuHz = 50 * 1000 * 1000;  // 50MHz
+
+const uint64_t kClockFreqPeripheralHz = 50 * 1000 * 1000;  // 50MHz
+
+const uint64_t kClockFreqUsbHz = 48 * 1000 * 1000;  // 50MHz
```
Comment (sriyerg) [inline sw/device/lib/arch/device_fpga_nexysvideo.c:18]: nit: comment reads 50MHz (here and below)

Comment (tjaychen) [reply sw/device/lib/arch/device_fpga_nexysvideo.c]: ah thanks for catching that

File: sw/device/lib/arch/device.h [hunk 2]
```diff
@@ -51,9 +51,19 @@ typedef enum device_type {
...
-extern const uint64_t kClockFreqHz;
+extern const uint64_t kClockFreqCpuHz;
+
+/**
+ * The peripheral clock frequency of the device, in hertz.
+ */
+extern const uint64_t kClockFreqPeripheralHz;
+
+/**
+ * The usb clock frequency of the device, in hertz.
```
Comment (mcy) [inline sw/device/lib/arch/device.h:64]: usb -> USB

File: sw/device/lib/arch/device.h [hunk 3]
```diff
@@ -51,9 +51,19 @@ typedef enum device_type {
 extern const device_type_t kDeviceType;
 
 /**
- * The clock frequency of the device, in hertz.
+ * The cpu clock frequency of the device, in hertz.
```
Comment (mcy) [inline sw/device/lib/arch/device.h:54]: cpu -> CPU

File: sw/device/lib/arch/device.h [hunk 4]
```diff
@@ -51,9 +51,19 @@ typedef enum device_type {
...
  */
-extern const uint64_t kClockFreqHz;
+extern const uint64_t kClockFreqCpuHz;
+
+/**
+ * The peripheral clock frequency of the device, in hertz.
```
Comment (mcy) [inline sw/device/lib/arch/device.h:59]: Would you mind elaborating on this comment? What is the "peripheral clock", exactly? Same goes for the others (though they're mildly more obvious I feel...

File: sw/device/lib/arch/device.h (hunk 2 above)
Comment (tjaychen) [reply sw/device/lib/arch/device.h]: updated

File: sw/device/lib/arch/device.h (hunk 3 above)
Comment (tjaychen) [reply sw/device/lib/arch/device.h]: updated

File: sw/device/lib/arch/device.h (hunk 4 above)
Comment (tjaychen) [reply sw/device/lib/arch/device.h:60]: i added some description, can you let me know it's clear enough

**Review by lenary — APPROVED** [review]
LGTM

**Comment by lenary** [comment]
@tjaychen This is ready for merging, and doing so before #3211 is probably easiest, right?

**Comment by tjaychen** [comment]
> @tjaychen This is ready for merging, and doing so before #3211 is probably easiest, right?

yes sounds good, let me do that. I was hoping to merge #3211 first, but that's probably not realistic :)

**Pull request merged**